core/thread/src/TThread.cxx

```diff
@@ -350,9 +350,7 @@ void TThread::Init()
    {
      R__LOCKGUARD(gGlobalMutex);
      if (!ROOT::gCoreMutex) {
-        // To avoid dead locks, caused by shared library opening and/or static initialization
```

Comment:
Why is this comment no longer applicable?

Reply:
I had thought this was related to some of the platform-specific stuff in TTHREAD_TLS_DECL and that it's not an issue if ```static thread_local``` can be used across the board.

Reply:
See commit bd1894b2bc8306c789557ddf12266c9af6259046.
`tls_get_addr_tail` is not a ROOT lock/routine but a system one. (and, not 100% sure, but it might be involved in the (compiler's) implementation of static initialization)

Reply:
Do you know which version of ```define TTHREAD_TLS``` was being used for that stack trace? (if it was already using ```thread_local``` then I agree there's no reason the situation should be different now)

Reply:
I am pretty sure it was the default `TTHREAD_TLS` (i.e. ROOT's workaround).